### PR TITLE
Add C# 9 tests for 8 rules

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Related to #3668

S3343 CallerInformationParametersShouldBeLast.cs
S4462 CallToAsyncMethodShouldNotBeBlocking.cs
S3247 CastShouldNotBeDuplicated.cs
S4830 CertificateValidationCheck.cs
S3928 CheckArgumentException.cs
S1451 CheckFileLicense.cs
S100, S101 ClassAndMethodName.cs

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        {
+            get
+            {
+                return (_, _, _, _) => true;   //False negative
+            }
+        }
+
+        #region helpers
+
+        static HttpWebRequest CreateRQ()
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs:120]: now that I am looking at this, I could extract in a common file and pass that file to both pre-C#9 and C#9 tests

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/CallerInformationParametersShouldBeLastTest.cs [hunk 2]
```diff
@@ -36,6 +36,15 @@ public void CallerInformationParametersShouldBeLast()
                 new CallerInformationParametersShouldBeLast());
         }
 
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void CallerInformationParametersShouldBeLast_CSharp9()
+        {
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/CallerInformationParametersShouldBeLastTest.cs:42]: pedantic: can use arrow method

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/ClassAndMethodNameTest.cs [hunk 3]
```diff
@@ -71,6 +82,15 @@ public void MethodName()
                 ParseOptionsHelper.FromCSharp8);
         }
 
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void MethodName_CSharp9()
+        {
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/ClassAndMethodNameTest.cs:88]: pedantic: can use arrow method

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallToAsyncMethodShouldNotBeBlocking.CSharp9.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class Program
+    {
+        public Task Foo(Task<string> task)
+        {
+            Action<Task<string>, object> continuation;
+            return task.ContinueWith(state: null, continuationAction: (Task<string> _, object _) =>
+            {
+                Task<int> anotherTask = null; // Pretend to compute something
+                var b = anotherTask.Result; // Noncompliant, this task is not safe inside ContinueWith
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallToAsyncMethodShouldNotBeBlocking.CSharp9.cs:15]: Theoretically we can call `.Result` also when using top-level statements. Shouldn't we add a test for that?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallerInformationParametersShouldBeLast.CSharp9.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void OuterMethod() {
+            void Local1([CallerFilePath] string callerFilePath = "") { }
+            void Local2(string other, [CallerFilePath] string callerFilePath = "") { }
+            void Local3([CallerFilePath] string callerFilePath = "", string other = "") { } // FN
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallerInformationParametersShouldBeLast.CSharp9.cs:10]: I think we could also add a test case with local functions in the context of top-level functions.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs [hunk 6]
```diff
@@ -0,0 +1,149 @@
+﻿
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs:1]: pedantic: there are a few extra white spaces in this file (e.g. extra empty line on top, trailing whitespace on line 21, 41)

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
Overall it looks good, just a few small comments

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs (hunk 6 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.CSharp9.cs:1]: thanks, I didn't have the Trailing Whitespace visualizer installed in the preview IDE :)

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**